### PR TITLE
POC initial

### PR DESCRIPTION
**Pull request by Elyash**

### CODE REVIEW

File: example/test_tetanus.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    """A Giraffe spec tests base.
+
+    Args:
+        giraffe: Giraffe instance.
+    """
+
+    _giraffe: Giraffe = None # type: ignore
+
+    @classmethod
+    def setup_class(cls, connections: List[Giraffe]) -> None:
```
Comment (yelly) [inline example/test_tetanus.py:26]: This is an example that the marking is confusing.
You require the inheriting class to mark.

File: example/test_tetanus.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        "The test send packets and expect them back."""
+
+        zebra, *_ = connections
+        await zebra.send_and_receive(TestsSpecTetanus._giraffe.get_ip(), self._echo_port)
+
+    @pytest.mark.lego('zebra and elephant')
+    async def test_multi_send_and_recv(self, connections): # type: ignore
+        """The test send packets from multiple components and
+            expect them back.
+        """
```
Comment (yelly) [inline example/test_tetanus.py:82]: The summary should always fit on one line.
Fix everywhere.

File: lego_manager/lego_manager.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+import contextlib
+import rpyc
+
+
+class LegoManager(rpyc.Service):
+    """
+    Setup and tests manager.
+    Should manage the permissions for tests to run on setup.
+    Stores its database using Rest API.
```
[the remark this answers is not in the document]
Comment (ykrinsky) [reply lego_manager/lego_manager.py:24]: I've added more detailed docstring at the top of the module explaining on lego manager.
I've mentioned LegoManager's DB isn't implemented yet, and this feature would be implemented in the future.

File: lego/setup.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+"""This file describes the lego plugin setup info."""
```
[the remark this answers is not in the document]
Comment (ykrinsky) [reply lego/setup.py:1]: Currently, the repo is divided to 3 parts:
1. Example - A usage example of lego3.
2. lego - Contains lego pytest plugin, and the base components and components factory.
3. lego manager.
They are different parts and, in a way, unrelated to each other, therefore we've decided to put setup.py at the top level of lego directory.

**Review by ykrinsky — APPROVED** [review]
Approve merge to master as we decided it would be better if all the code will be in master and not in poc_initial.

**Pull request merged**